src/grammar.json

```diff
@@ -0,0 +1,2518 @@
+{
```

Comment:
It is so those that want to install the parser do not need to generate from the grammar file, which nvim-treesitter  (neovim plugin) does. This way, all one needs to install this parser is a C compiler, which reduces time spent on installing and resources as well, at the cost of having this file in the repo. Diffs and PRs won't show changes to these files because of the .gitattributes file I added if that's a concern of yours

Reply:
I am for keeping only source code under source revision control.
Published NPM package will include all generated files.
Same for Cargo and other Package managing systems.
Keeping large generated files under source code revision control leading to confusion and mismatch IMHO

Reply:
nvim-treesitter, helix, emacs, nixpkgs, and whatever other editor/pkg management system almost always pulls the git version as oftentimes repos don't publish often enough to fix bugs/add features. I would like to say that the .gitattributes file prevents unnecessary noise in viewing diffs / changes online. See example here: 
![image](https://user-images.githubusercontent.com/29718261/222941691-604b05f8-c810-4bcb-b6dc-49e903e3faf6.png)

So, I'd like to counter the idea that this would lead to confusion/mismatches, provided that PRs changing the grammar also update the generated files (ideally with a "generate" commit of sorts)

Reply:
Git is not a very good "package manager". If some systems can't download / untar NPM package we can create separate branch "package" or "release" or something, similar to "gh-pages" where we will "publish" releases of generated C, H, JSON files (no source) and that would be a "pure-man package manager". How about this?

Reply:
I don't particularly like that as any sources that leverage tree-sitter packages have it configured to either generate from npm if there's no generated files, or just simple compile parser.c if it's present.

Just doing a quick check in nvim-treesitters parsers.lua list, about 30 of the ~200 languages we support require this generation process. So for the most part, grammars contain these files so people on lower end machines won't struggle as much and the only dependency is a C99 compiler then (making it much more cross-platform/universally available)..

Of course it's your final call as it's your repo, but to me it'd have far more benefits than downsides to add them here